rotkehlchen/tasks/history_base_entries.py

```diff
@@ -0,0 +1,79 @@
+import copy
+import logging
+from typing import TYPE_CHECKING, Iterator, List, Set, Tuple
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants.assets import A_USD
+from rotkehlchen.db.filtering import DBExcludeValuesFilter, DBStringFilter, HistoryEventFilterQuery
+from rotkehlchen.errors import NoPriceForGivenTimestamp
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.price import PriceHistorian
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.typing import Timestamp
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def get_base_entry_rows_missing_prices(
+    filter_query: HistoryEventFilterQuery,
+    db: 'DBHandler',
+    ignore_list: List[str],
+) -> Tuple[Iterator[Tuple[str, FVal, Asset, Timestamp]], int]:
+    """
+    Get the list of rows as a generator that need to update their prices
+    taking into account the identifiers that should be ignored at the ignore_list.
+    It returns both a generator with the rows that need to be updated and the amount
+    of entries in that generator.
+    """
+    # Use a deepcopy to avoid mutations in the filter query if it is used later
+    new_filter_query = copy.deepcopy(filter_query)
+    new_filter_query.filters.append(DBStringFilter(and_op=True, column='usd_value', value='0'))
+    new_filter_query.filters.append(
+        DBExcludeValuesFilter(
+            and_op=True,
+            column='identifier',
+            values=ignore_list,
+        ),
+    )
+    entries_missing_prices, count = db.iterate_rows_missing_prices_in_base_entries(
+        filter_query=new_filter_query,
+    )
+    return entries_missing_prices, count
+
+
+def query_missing_prices(
+    rows: Iterator[Tuple[str, FVal, Asset, Timestamp]],
+    db: 'DBHandler',
+    ignore_list: Set[str],
```

Comment:
Sorry but no Yabir, this entire code with doing all these extra gymnastics just to keep the iterator is terribly ugly and unnecessarily complicated. I told you in discord to not do this.

Since I last checked you seem to have:

1. Added a new exclude filter to the DB query
2. Messed up ignore_list with ignore_set. Decide. Is it a list? Is it a set? Treat it as one all across the code.
3. You are passing a set argument through multiple functions many levels deep, turning it into a list some times and then mutate it. Mutating an argument is a terrible, terrible practise. It's hard to follow, you have no guarantees on the code, you have to know the argument is mutated and you will forget that it does so in a week. Imagine someone new reading the code.

All this for what? To just keep returning an iterator? Please keep it simple, keep it readable and keep proper coding interfaces and let's close this PR.

Please do the following:
1. Query X history events with 0 value. I already told you to limit the number of event queried in a single task. Look at what we do for transaction receipts.
2. For each of the X history events query the price. If the price query fails note it in the task manager's state. Or in whatever object's state. This is where you have to keep proper interfaces and abstractions. Passing a set to be mutated by some random functions is BAD, BAD, BAD, BAD. A developer should not have to read the code to kind of guess where things should happen. Keep proper abstractions and interfaces.
3. If the price query succeeds save it in the DB